File: 2_gyak/Liskov/Java/LiskovraFigyel.java
public class Liskovra_figyel {

	class Madarak{

	}

	class RepuloMadar extends Madarak{

		void repul(){
			System.out.println( this.toString() + " ... repul ... ");
		}

	}


	class Pingvin extends Madarak{

	}
	
	class Sas extends RepuloMadar{

	}
	
	public static void main(String[] args) {
		Liskovra_figyel liskov = new Liskovra_figyel();
		
		Pingvin pingvin = liskov.new Pingvin();
		Sas sas = liskov.new Sas();
		
	//	pingvin.repul();
		sas.repul();
	}
}

File: 2_gyak/Liskov/C++/LiskovraFigyel.cpp
#include <iostream>

using namespace std;

class Madarak {
		
};

class RepuloMadar: public Madarak {

public:
	void repul(){
		cout << "repul ..." << endl;
	}

};

class Sas: public RepuloMadar {

};

class Pingvin : public Madarak {

};

int main ( int argc, char **argv ){
     
	Pingvin pingvin;
	Sas sas;

//	pingvin.repul();
	sas.repul();

	return 0;
}
